### PR TITLE
[4.0] Add extension name/id to uninstall message if extension is locked

### PR DESCRIPTION
**Pull request by ReLater**
### Summary of Changes
- Change lang string COM_INSTALLER_UNINSTALL_ERROR_LOCKED_EXTENSION
- Use sprintf to display extension name and id

### Testing Instructions
- 4.0 nightly build of today (Sunday, 03 May 2020 02:00:34 UTC)
- Go to extensions manager (`administrator/index.php?option=com_installer&view=manage`)
- Select many locked extensions.
- Click `Uninstall`.

### Additional comment:
- I can't test changes in `libraries/src/Console/ExtensionRemoveCommand.php` in my current environment.

### CODE REVIEW

**Comment by richard67** [comment]
@ReLater Could you translate the extension names in the messages? I think if you just use `Text::_($row->name)` instead of `$row->name` it should work. Then you could change your language sting to "Locked extensions cannot be uninstalled: Extension name %s, ID %s.", and people would see it is the name which they can see e.g. in "Extensions: Manage".

**Comment by infograf768** [comment]
if you change to variables please number them. That helps TT when multiple variables.

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on e114d0bf6cdbebe332f1c5a3b0fefb5144ac34a4<br /><br />Confirm Actual result.
Confirm After Patch, see screenshot, but I don't know what to do with sprintf
Installed: 4.0 nightly build of today (Sunday, 03 May 2020 02:00:34 UTC)
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28916">issues.joomla.org/tracker/joomla-cms/28916</a>.</sub>
pr28916

**Comment by ChristineWk** [comment]
pr28916

File: administrator/language/en-GB/com_installer.ini [hunk 1]
```diff
@@ -249,7 +249,7 @@ COM_INSTALLER_TYPE_TYPE_TEMPLATE="template"
 COM_INSTALLER_UNABLE_TO_FIND_INSTALL_PACKAGE="Unable to find install package"
 COM_INSTALLER_UNABLE_TO_INSTALL_JOOMLA_PACKAGE="The Joomla package cannot be installed through the Extension Manager. Please use the <a href='%s'>Joomla! Update</a> component to update Joomla."
 COM_INSTALLER_UNINSTALL_ERROR="Error uninstalling %s."
-COM_INSTALLER_UNINSTALL_ERROR_LOCKED_EXTENSION="Locked extensions cannot be uninstalled."
+COM_INSTALLER_UNINSTALL_ERROR_LOCKED_EXTENSION="Locked extensions cannot be uninstalled. %s, ID %s."
```
Comment (brianteeman) [inline administrator/language/en-GB/com_installer.ini]:
```suggestion
COM_INSTALLER_UNINSTALL_ERROR_LOCKED_EXTENSION="Locked extension cannot be uninstalled. %s, ID %s."
```

**Comment by ReLater** [comment]
@richard67 
> I think if you just use Text::_($row->name) instead of $row->name it should work.

I've tested that before I posted pr. You would have to load all relevant language files first. Complete overhead for this little feature.

That's the reason why I've added the ID.

**Comment by richard67** [comment]
Last change was just a small language string change, so previous test result is still valid.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 08e9844d6dd7470a69abcb4f03b4313407c61143<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28916">issues.joomla.org/tracker/joomla-cms/28916</a>.</sub>

**Comment by ReLater** [comment]
@ChristineWk 

> but I don't know what to do with sprintf

Ignore that comment below headline "Summary of Changes". It's just an abstract of what I've done in codes. "Code uses sprintf now."

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28916">issues.joomla.org/tracker/joomla-cms/28916</a>.</sub>

**Comment by ReLater** [comment]
@infograf768 

> if you change to variables please number them. That helps TT when multiple variables.

Sorry. I don't understand that comment.

**Comment by brianteeman** [comment]
What he means is that instead of having

> The extension %s which is a %s cannot be found

You have 

> The extension %1s which is a %2s cannot be found

This makes it easier for translators where the order of parts is not the same

**Comment by ChristineWk** [comment]
@ReLater 
ah, I see now: "Code uses sprintf now." > "Locked extension cannot be uninstalled. **%s, ID %s.**"

File: administrator/language/en-GB/com_installer.ini [hunk 2]
```diff
@@ -249,7 +249,7 @@ COM_INSTALLER_TYPE_TYPE_TEMPLATE="template"
 COM_INSTALLER_UNABLE_TO_FIND_INSTALL_PACKAGE="Unable to find install package"
 COM_INSTALLER_UNABLE_TO_INSTALL_JOOMLA_PACKAGE="The Joomla package cannot be installed through the Extension Manager. Please use the <a href='%s'>Joomla! Update</a> component to update Joomla."
 COM_INSTALLER_UNINSTALL_ERROR="Error uninstalling %s."
-COM_INSTALLER_UNINSTALL_ERROR_LOCKED_EXTENSION="Locked extensions cannot be uninstalled."
+COM_INSTALLER_UNINSTALL_ERROR_LOCKED_EXTENSION="Locked extension cannot be uninstalled. %s, ID %s."
```
Comment (ReLater) [inline administrator/language/en-GB/com_installer.ini]:
```suggestion
COM_INSTALLER_UNINSTALL_ERROR_LOCKED_EXTENSION="Locked extension cannot be uninstalled. %1s, ID %2s."
```

**Comment by ReLater** [comment]
Decide yorself if we need new tests after last change https://github.com/joomla/joomla-cms/pull/28916/commits/246dea7cf34c56158d4fbdef453a21d2f087a406

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 246dea7cf34c56158d4fbdef453a21d2f087a406<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28916">issues.joomla.org/tracker/joomla-cms/28916</a>.</sub>

**Comment by richard67** [comment]
Back to pending. One more test needed.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28916">issues.joomla.org/tracker/joomla-cms/28916</a>.</sub>

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 246dea7cf34c56158d4fbdef453a21d2f087a406<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28916">issues.joomla.org/tracker/joomla-cms/28916</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28916">issues.joomla.org/tracker/joomla-cms/28916</a>.</sub>

**Comment by infograf768** [comment]
This gives
Screen Shot 2020-05-04 at 08 41 17

Would'nt it be better to get
Screen Shot 2020-05-04 at 08 45 36

String
`COM_INSTALLER_UNINSTALL_ERROR_LOCKED_EXTENSION="The \"%1s\" (ID %2s) is a locked extension and cannot be uninstalled."`

What do you think?

**Comment by richard67** [comment]
Hmm ... either leave the `The ` at the beginning, so it is `COM_INSTALLER_UNINSTALL_ERROR_LOCKED_EXTENSION="\"%1s\" (ID %2s) is a locked extension and cannot be uninstalled."`

Or change to `COM_INSTALLER_UNINSTALL_ERROR_LOCKED_EXTENSION="The extension \"%1s\" (ID %2s) is locked and cannot be uninstalled."`

**Comment by infograf768** [comment]
OK for 2nd proposal. It's definitely better phrasing than the existing one.

**Comment by infograf768** [comment]
@ReLater 
Can you modify the string please?

**Comment by ReLater** [comment]
Ah, come on! Feel free to take over.

**Comment by richard67** [comment]
@ReLater Please, restore your branch and do that little change:
`COM_INSTALLER_UNINSTALL_ERROR_LOCKED_EXTENSION="The extension \"%1s\" (ID %2s) is locked and cannot be uninstalled."`

If you really don't want, restore at least the branch so that I can pick your commit, then you will still be counted as co-author in a new PR.

**Pull request closed without merging**